sw/device/silicon_creator/lib/crypto/tests/rsa_3072_verify_set_testvectors.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import sys
+
+import hjson
+
+'''
+Read in a JSON test vector file, convert the test vector to C constants, and
+generate a header file with these test vectors.
+'''
+
+HEADER_PREFIX = '''// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// AUTOGENERATED. Do not edit this file by hand.
+// See the crypto/tests README for details.
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CRYPTO_TESTS_RSA_3072_VERIFY_TESTVECTORS_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CRYPTO_TESTS_RSA_3072_VERIFY_TESTVECTORS_H_
+
+#include "sw/device/silicon_creator/lib/crypto/rsa_3072/rsa_3072_verify.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// A test vector for RSA-3072 verify (message hashed with SHA-256)
+typedef struct rsa_3072_verify_test_vector_t {
+  rsa_3072_public_key_t publicKey;  // The public key
+  rsa_3072_int_t signature;         // The signature to verify
+  char *msg;                        // The message
+  size_t msgLen;                    // Length (in bytes) of the message
+  bool valid;                       // Expected result (true if signature valid)
+  char *comment;                    // Any notes about the test vector
+} rsa_3072_verify_test_vector_t;
+
+'''
+
+
+HEADER_END = '''
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif  // __cplusplus
+
+#endif  // OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CRYPTO_TESTS_RSA_3072_VERIFY_TESTVECTORS_H_
+'''
+
+# Number of 32-bit words in a 3072-bit number
+RSA_3072_NUMWORDS = int(3072 / 32)
+
+
+def rsa_3072_int_to_words(x):
+    '''Convert a 3072-bit integer to a list of 32-bit integers (little-endian).'''
+    out = []
+    for i in range(RSA_3072_NUMWORDS):
+        out.append((x >> (i * 32)) & (2**32 - 1))
```

Comment:
I think a somewhat clearer way to write this is
```python
for _ in range(...):
  out.append(x & (2**32 - 1))
  x >>= 32
```
I mostly say this because the `i*32` seriously threw me off.

Reply:
I'd also consider hoisting the `0xffffffff` mask out of the loop and would personally use `(1 << 32)` instead of `2**32` (it's technically faster, not that anyone is going to care, and it's probably a bit more obvious what's going on). (Super-nitty: if you're ready to merge this change otherwise, no need to re-push!)